torch/fx/symbolic_trace.py

```diff
@@ -307,13 +307,140 @@ def forward(*args, **kwargs):
             # Seems to be a mypy limitation: https://github.com/python/mypy/issues/2427
             torch.nn.Module.__getattr__ = module_getattr_wrapper  # type: ignore
             torch.nn.Module.__call__ = module_call_wrapper
+
+            orig_fns = _patch_wrapped_functions()
+
             self.create_node('output', 'output', (self.create_arg(fn(*args)),), {},
                              type_expr=fn.__annotations__.get('return', None))
         finally:
+            _unpatch_wrapped_functions(orig_fns)
             torch.nn.Module.__call__ = orig_call
             torch.nn.Module.__getattr__ = orig_getattr  # type: ignore
         return self.graph
 
+class _GlobalDictWrapper:
+    """
+    Hashable wrapper to use global dicts from Python frames
+    as keys.
+    """
+    def __init__(self, global_dict):
+        self.global_dict = global_dict
+
+    def __hash__(self):
+        return id(self.global_dict)
+
+    def __eq__(self, other):
+        return id(self.global_dict) == id(other.global_dict)
+
+    def __getitem__(self, key):
+        return self.global_dict[key]
+
+    def __setitem__(self, key, value):
+        self.global_dict[key] = value
+
+# This maps from global namespace dictionaries to
+# an ordered set of functions that should be patched
+# for wrapping during symbolic tracing
+_wrapped_fn_patch_table : Dict[_GlobalDictWrapper, Dict['str', Any]] = {}
+
+def _call_wrapped_func(orig_fn, args, kwargs):
+    """
+    Given an ``orig_function`` to invoke, search the args and kwargs for
+    a Proxy object. If there is one, emit a ``call_function`` node to
+    preserve the call to this leaf function directly. Otherwise, just
+    return the results of this function call, as this function is not
+    being traced.
+    """
+    proxy = None
+
+    def find_proxy(x):
+        nonlocal proxy
+        if isinstance(x, Proxy):
+            proxy = x
+
+    map_aggregate(args, find_proxy)
+    map_aggregate(kwargs, find_proxy)
+
+    if proxy is not None:
+        return proxy.tracer.create_proxy('call_function', orig_fn, args, kwargs)
+    else:
+        return orig_fn(*args, **kwargs)
+
+def _patch_wrapped_functions() -> Dict[_GlobalDictWrapper, Dict['str', Any]]:
+    """
+    Go through ``_wrapped_fn_patch_table`` and, for each frame object, wrap
+    the listed global functions in the `call_wrapped_func` wrapper. Returns
+    a map from frame to dict mapping name to the original global function.
+    """
+    orig_fns : Dict[_GlobalDictWrapper, Dict['str', Any]] = {}
```

Comment:
This can just be `List[callable]` of the original functions in the same order as the wrap list.